fondat/aws/bedrock/resources/agent.py

```diff
@@ -0,0 +1,262 @@
+        enableTrace: bool = False,
+        endSession: bool = False,
+        guardrailConfiguration: dict | None = None,
+        idleSessionTTLInSeconds: int | None = None,
+        inlineSessionState: dict | None = None,
+        knowledgeBases: list[dict] | None = None,
+        orchestrationType: str | None = None,
+        promptOverrideConfiguration: dict | None = None,
+        streamingConfigurations: dict | None = None,
+    ) -> Mapping[str, Any]:
+        """
+        Invoke an inline Amazon Bedrock agent.
+
+        Required:
+        foundationModel: the underlying model to orchestrate
+        instruction: what the agent should do  
+        sessionId:   unique session identifier  
+        inputText:   user input to send to the agent
+
+        Optional: you can pass any of the other inline-agent config blocks below.
+        """
+        params: dict[str, Any] = {
+            "agentId": self._id,
+            "foundationModel": foundationModel,
+            "instruction": instruction,
+            "sessionId": sessionId,
+            "inputText": inputText,
+        }
+        if actionGroups:
+            params["actionGroups"] = actionGroups
+        if agentCollaboration:
+            params["agentCollaboration"] = agentCollaboration
+        if agentName:
+            params["agentName"] = agentName
+        if bedrockModelConfigurations:
+            params["bedrockModelConfigurations"] = bedrockModelConfigurations
+        if collaboratorConfigurations:
+            params["collaboratorConfigurations"] = collaboratorConfigurations
+        if collaborators:
+            params["collaborators"] = collaborators
+        if customOrchestration:
+            params["customOrchestration"] = customOrchestration
+        if customerEncryptionKeyArn:
+            params["customerEncryptionKeyArn"] = customerEncryptionKeyArn
+        if enableTrace:
+            params["enableTrace"] = True
+        if endSession:
+            params["endSession"] = True
+        if guardrailConfiguration:
+            params["guardrailConfiguration"] = guardrailConfiguration
+        if idleSessionTTLInSeconds is not None:
+            params["idleSessionTTLInSeconds"] = idleSessionTTLInSeconds
+        if inlineSessionState:
+            params["inlineSessionState"] = inlineSessionState
+        if knowledgeBases:
+            params["knowledgeBases"] = knowledgeBases
+        if orchestrationType:
+            params["orchestrationType"] = orchestrationType
+        if promptOverrideConfiguration:
+            params["promptOverrideConfiguration"] = promptOverrideConfiguration
+        if streamingConfigurations:
+            params["streamingConfigurations"] = streamingConfigurations
+
+        async with runtime_client(self.config_runtime) as client:
+            with wrap_client_error():
+                return await client.invoke_inline_agent(**params)
+
+    @property
+    def versions(self) -> VersionsResource:
```

Comment:
Should we return a resource or the representation of the resource (i.e. a `Version` class) ?